### PR TITLE
Development

### LINKED ISSUE #11: Merge Predicate and Value
Some combinations are useles: `Eq $ Range "0" "1"`, `Lt $ Range "foo" "bar"` 
It's better to make `Range` `Predicate`. And `data Value = Tag String | Text String`. 
Inputs like `>foo:12..123` should be incorrect.

### LINKED ISSUE #21: Import docs to pursuit?
It'd be really awesome if these docs could be on http://pursuit.purescript.org/ - I think the setup is a simple github oauth request approval.

### PR DESCRIPTION
**Pull request by cryogenian**
fixes #11 
fixes #21 
- Pursuit
- Dependencies are updated
- Added some comments 
- Removed unused by `SlamSearch` syntax symbols from reserved symbols list. (Except `(` and `)`)
- Fixed https://slamdata.atlassian.net/browse/SD-1081
- Queries like `>foo:bar` or `foo:!bar..baz` couldn't be parsed. (They used to be parsed to incorrect `SearchQuery`) 
- Removed `qualified`, changed formatting a bit. 

@garyb Please review

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -5,4 +5,12 @@ install:
...
-  - gulp test
+  - gulp test --psc-rts-opts -N4
+after_success:
+- >-
+  test $TRAVIS_TAG &&
+  ./node_modules/.bin/psc-publish > .pursuit.json &&
+  curl -X POST http://pursuit.purescript.org/packages \
+    -d @.pursuit.json \
+    -H 'Accept: application/json' \
+    -H "Authorization: token ${GITHUB_TOKEN}"
```
Comment (garyb) [inline .travis.yml:16]: When we add this we also need to add the `GITHUB_TOKEN` in the build settings for the repo in travis. There's [some info here](http://pursuit.purescript.org/help) about generating the token under "Submitting packages from a script" - no need to do it for this repo though, I've set it up already.

File: bower.json [hunk 2]
```diff
@@ -19,12 +19,14 @@
...
     "purescript-monoid": "^0.3.2",
     "purescript-parsing": "^0.7.1",
     "purescript-semirings": "^0.2.0",
-    "purescript-strings": "^0.7.0",
-    "purescript-strongcheck": "^0.14.4"
+    "purescript-strings": "^0.7.1"
+  },
+  "devDependencies": {
+    "purescript-console": "^0.1.1",
+    "purescript-strongcheck": "^0.14.5"
```
Comment (garyb) [inline bower.json]: I think strongcheck is a "real" dependency rather than a dev dependency, as we declare `Arbitrary` instances for it under `src`.

**Comment by garyb** [comment]
:+1: aside from the strongcheck dependency.

File: bower.json (hunk 2 above)
Comment (jonsterling) [inline bower.json]: agreed

**Comment by cryogenian** [comment]
strongcheck dependency is moved, commits are squashed

**Pull request merged**